### DISCUSSION TITLE
Coral USB TPU and Python

### DISCUSSION THREAD

**Discussion by billynlazer** [roflcoopter/viseron#654] [Q&A] [answered]
Newbie here, so apologies if this question has an obvious answer...

I've muddled through and managed to install Viseron in Docker on Ubuntu, and am interested in buying a Coral USB TPU to get object detection working properly. Before I drop the dough on buying the Coral USB Accelerator, I thought I'd try installing some of the prerequisites listed here https://coral.ai/docs/accelerator/get-started/. All went well until I got to the  Python part, where it failed 

The following packages have unmet dependencies:
 python3-pycoral : Depends: python3-tflite-runtime (= 2.5.0.post1) but it is not going to be installed
                   Depends: python3 (< 3.10) but 3.10.6-1~22.04 is to be installed

ie I have Python 3.10 installed, but need Python 3.9 or less.

Now from my rudimentary understanding of Docker, I understand that the Viseron docker container actually has its own Python 3.9 and so as long as I have the Coral USB plugged in and the line -v /dev/bus/usb:/dev/bus/usb --privileged as a volume, then Viseron should make use of the USB Coral and use the Python 3.9 it gets from the docker container - ie it should run without me going to the trouble of downgrading 3.10 down to 3.9.

Is my understanding correct?

**Comment by roflcoopter** [comment, marked as answer]
That is correct!
Its the beauty of Docker, it runs in sort of a light weight virtual machine so you dont have to worry about dependencies.
The Docker container has everything installed to run the Coral.

**Reply by billynlazer** [reply]
Wonderful! Thank you very much! :)
